### ISSUE TITLE
Domains Table: Filter on domain group and change filter behavior for status

### ISSUE THREAD

**Issue by zandercymatics** [cisagov/manage.get.gov#2440]
### Story

Domain groups are meant to act like a label on a domain. These will be set by admin users to "group" domains together to help with their own internal organization. For instance, some users may group domains that have a particular manager or who expire within the same month. Just like in github where we can filter by a label, the goal is to let users filter by any set of domain groups (labels).

When revising the design for the Domain Table, it was found that our existing filter design doesn't handle the large number of options present for domain group very well. For consistency, we should match this new filter design on status as well.

As an org manager
I want the ability to filter on domain groups
so that I can narrow down my search

### Acceptance Criteria

- [ ] The domain group column is filterable on domain group
- [ ] The remaining filter buttons are refactored to behave like the domain group button
- [ ] When a filter button is clicked (such as status), the same filter menu appears and it opens the associated status accordion 

### Additional Context

[Consult this figma](https://www.figma.com/design/xFtGLHVrhp0lvwh0gYWnbx/.gov-Designs-CISA?node-id=5380-23248&t=oNwL0b23uBsGuz0l-0) (*TBD*).

As a reminder: Domain groups can be up to 320 characters

Refinement note: this ticket is awaiting further design deliberation by @AnnaGingle and @gabydisarli 

### Issue Links
⛔ blocked by 
  - #2395 
 
🔄 Relates to
- #2396
- #2453

**Comment by AnnaGingle** [comment]
@Katherine-Osos @gabydisarli We decided to remove filtering from this group of work so we can design something that works for all of our tables and potential filters.

**Comment by gabydisarli** [comment]
> @Katherine-Osos @gabydisarli We decided to remove filtering from this group of work so we can design something that works for all of our tables and potential filters.

New issue created: #2453

**Comment by PaulKuykendall** [comment]
Closing this ticket, but leaving it linked. A new ticket will be created by design, per #2453 .
